content/actions/hosting-your-own-runners/using-self-hosted-runners-in-a-workflow.md

```diff
@@ -70,9 +70,9 @@ These labels operate cumulatively, so a self-hosted runner’s labels must match
 
 When routing a job to a self-hosted runner, {% data variables.product.prodname_dotcom %} looks for a runner that matches the job's `runs-on` labels:
 
-1. {% data variables.product.prodname_dotcom %} first searches for a runner at the repository level, then at the organization level{% ifversion ghes or ghae %}, then at the enterprise level{% endif %}. 
-2. The job is then sent to the first matching runner that is online and idle.
-   - If all matching online runners are busy, the job will queue at the level with the highest number of matching online runners.
-   - If all matching runners are offline, the job will queue at the level with the highest number of matching offline runners.
-   - If there are no matching runners at any level, the job will fail.
+- {% data variables.product.prodname_dotcom %} first searches for an online and enabled runner at the repository level, then at the organization level{% ifversion ghes or ghae %}, then at the enterprise level{% endif %}.
+- If {% data variables.product.prodname_dotcom %} doesn't find an online and enabled runner at any level, the job is queued to all levels and waits for any runner from any level to come online and pickup the job.
    - If the job remains queued for more than 24 hours, the job will fail.
+- If {% data variables.product.prodname_dotcom %} finds an online and enabled runner (preferred runner) at a certain level, the job is then sent to the preferred runner.
+    - If the job is not picked up by the preferred runner within 60 seconds after sending the job, {% data variables.product.prodname_dotcom %} will try send the same job to all other levels as well.
+    - If the job remains queued for more than 24 hours, the job will fail.
```

Comment:
How about we put the match condition first, so we can sum up the queuing and timeout description once at the end. Does this ordering/wording make sense?


```suggestion
- {% data variables.product.prodname_dotcom %} first searches for an online and enabled runner at the repository level, then at the organization level{% ifversion ghes or ghae %}, then at the enterprise level{% endif %}.
- If {% data variables.product.prodname_dotcom %} finds an online and enabled runner at a certain level that matches the job's `runs-on` labels, the job is then assigned and sent to the runner.
    - If the runner doesn't pick up the assigned job within 60 seconds, {% data variables.product.prodname_dotcom %} starts searching again at all levels for an online and enabled runner.
- If {% data variables.product.prodname_dotcom %} doesn't find an online and enabled runner at any level, the job is queued to all levels and waits for any runner from any level to come online and pick up the job.
   - If the job remains queued for more than 24 hours, the job will fail.
```